lib/components/Autosuggest/Autosuggest.docs.tsx

```diff
@@ -197,6 +195,127 @@ const docs: ComponentDocs = {
       },
     },
   ],
+  snippets: [
+    {
+      name: 'Standard',
+      code: (
+        <Autosuggest
+          id="fruit"
+          label="Fruit"
+          value={{ text: '' }}
+          onChange={() => {}}
+          suggestions={[
+            { text: 'Apples' },
+            { text: 'Bananas' },
+            { text: 'Carrots' },
+          ]}
+        />
+      ),
+    },
+    {
+      name: 'Grouped suggestions',
+      code: (
+        <Autosuggest
+          label="I like to eat"
+          id="grouped"
+          value={{ text: '' }}
+          onChange={() => {}}
+          onClear={() => {}}
+          suggestions={[
+            {
+              label: 'Fruit',
+              suggestions: [
+                { text: 'Apples' },
+                { text: 'Bananas' },
+                { text: 'Carrots' },
+              ],
+            },
+            {
+              label: 'Vegetables',
+              suggestions: [
+                { text: 'Broccoli' },
+                { text: 'Carrots' },
+                { text: 'Carrots' },
+              ],
+            },
+          ]}
+        />
+      ),
+    },
+    {
+      name: 'With mobile backdrop',
+      code: (
+        <Autosuggest
+          showMobileBackdrop
+          id="mobile"
+          label="Fruit"
+          value={{ text: '' }}
+          onChange={() => {}}
+          suggestions={[
+            { text: 'Apples' },
+            { text: 'Bananas' },
+            { text: 'Carrots' },
+          ]}
+        />
+      ),
+    },
+    {
+      name: 'With error',
+      code: (
+        <Autosuggest
+          label="I like to eat"
+          id="error"
+          value={{ text: '' }}
+          onChange={() => {}}
+          tone="critical"
+          message="You must make a selection"
+          suggestions={[
+            { text: 'Apples' },
+            { text: 'Bananas' },
+            { text: 'Carrots' },
+          ]}
+        />
+      ),
+    },
+    {
+      name: 'With description',
+      code: (
+        <Autosuggest
+          label="Fruit"
+          id="error"
+          value={{ text: '' }}
+          onChange={() => {}}
+          description="Select your favourite fruit to eat from the available suggestions."
+          suggestions={[
+            { text: 'Apples' },
+            { text: 'Bananas' },
+            { text: 'Carrots' },
+          ]}
+        />
+      ),
+    },
+    {
+      name: 'Location search',
```

Comment:
Should this be something like "With icon" instead? Like what we've done with `TextField`?